265_dec: modify the length of sublayer_hrd_params

wudping opened the pull request:
According to specification 7.4.3.1 and 7.4.3.2.1, the value of
sps_max_sub_layers_minus1 and vps_max_sub_layers_minus1 shall be
in the range of 0 to 6, inclusive, so sublayer_hrd_params[]'s length
should be at least MAXSUBLAYERS(7). Or else, decoding will crash
when decoding some clips.

hevc 8 bit decoding. [VIZ-6406](https://jira01.devtools.intel.com/browse/VIZ-6406?filter=79676)

Signed-off-by: wudping <dongpingx.wu@intel.com>

dongpingx-wu committed:
265_dec: modify the length of sublayer_hrd_params

According to specification 7.4.3.1 and 7.4.3.2.1, the value of
sps_max_sub_layers_minus1 and vps_max_sub_layers_minus1 shall be
in the range of 0 to 6, inclusive, so sublayer_hrd_params[]'s length
should be at least MAXSUBLAYERS(7). Or else, decoding will crash
when decoding some clips.

Signed-off-by: wudping <dongpingx.wu@intel.com>

wudping commented:
please help to review.

xuguangxin commented on codecparsers/h265Parser.h:
```diff
@@ -143,7 +143,7 @@ namespace H265 {
         bool low_delay_hrd_flag[6];
         uint8_t cpb_cnt_minus1[6]; //[0, 31]
 
-        SubLayerHRDParameters sublayer_hrd_params[6];
```
do we need change fixed_pic_rate_general_flag,  fixed_pic_rate_within_cvs_flag, elemental_duration_in_tc_minus1, low_delay_hrd_flag, cpb_cnt_minus1 seems they use same loop variable maxNumSubLayersMinus1

wudping replied:
okay, Thanks @xuguangxin .

dongpingx-wu committed:
265_parse: add deblocking filter value seting

According to spec 7.4.7.1, If slice_deblocking_filter_disabled_flag
is not present in slice, assign pps_deblocking_filter_disabled_flag
to it.

Signed-off-by: wudping <dongpingx.wu@intel.com>

wudping commented:
updated according to the comments, and set PPS's deblocking filter parameter to the SLICE's when the SLICE's deblocking parameter doesn't present. 
Please help to review.

xuguangxin commented on codecparsers/h265Parser.cpp:
```diff
@@ -1473,10 +1473,17 @@ bool Parser::parseSlice(const NalUnit* nalu, SliceHeader* slice)
+                slice->tc_offset_div2 = pps->pps_tc_offset_div2;
+            }
+        }
+        CHECK_RANGE(slice->beta_offset_div2, -6, 6);
```
better add CHECK_READ_SE and check this at the location when read happened. I mean pps->pps_beta_offset_div2 and slice->beta_offset_div2

wudping replied:
okay, Thanks @xuguangxin .

xuguangxin commented on codecparsers/h265Parser.cpp:
```diff
@@ -1406,10 +1391,6 @@ bool Parser::parseSlice(const NalUnit* nalu, SliceHeader* slice)
                 slice->num_ref_idx_l1_active_minus1 = pps->num_ref_idx_l1_default_active_minus1;
```
should we check pps->num_ref_idx_l1_default_active_minus1?

xuguangxin commented on codecparsers/h265Parser.cpp:
```diff
@@ -76,10 +85,12 @@
-    {                                       \
-        if ((var) < (min) || (var) > (max)) \
-            return false;                   \
+#define CHECK_RANGE(var, min, max)                                               \
```
could we use this macro for CHECK_READ_BITS and CHECK_READ_UE

wudping replied:
okay, Thanks @xuguangxin .

dongpingx-wu committed:
265_parse: check if the value in the range

Adding a ERROR log before 'return false', indicate that
read value is not in the range.

Signed-off-by: wudping <dongpingx.wu@intel.com>

wudping commented:
update according to the suggestions, please help to review.

xuguangxin commented on codecparsers/h265Parser.cpp:
```diff
@@ -24,6 +24,26 @@
 
+#define POWER32SUB2 4294967294
+
+#define POWER15 32768
```
please use (2<<15)

xuguangxin commented on codecparsers/h265Parser.cpp:
```diff
@@ -24,6 +24,26 @@
 #include <math.h>
 #include <stddef.h> //offsetof
 
+#define POWER32SUB2 4294967294
```
please use ((2<<32)-2)

xuguangxin commented on codecparsers/h265Parser.cpp:
```diff
@@ -99,15 +128,11 @@ namespace H265 {
+            CHECK_READ_UE(var->var##_max_num_reorder_pics[i], 0, var->var##_max_dec_pic_buffering_minus1[i]);    \
                                                                                                                  \
-            READ_UE(var->var##_max_latency_increase_plus1[i]);                                                   \
+            CHECK_READ_UE_LONG(var->var##_max_latency_increase_plus1[i], 0, POWER32SUB2);                        \
```
why we need CHECK_READ_UE_LONG, POWER32SUB2 is not long, right?

wudping commented:
updated according to suggestion, please help to review. thanks again.

wudping commented:
updated according to suggestions, please help to review..

xuguangxin commented:
merged as 125f35d8412252aa67efcb7f13737746a1299f1e
